src/components/metadata/panel.js

```diff
@@ -163,23 +149,9 @@ class MetadataPanel extends React.PureComponent {
     }
   }
 
-  showFieldPopup = ({ detail }) => {
-    let { id, type } = detail.target
-
-    this.setState({
-      fieldPopup: {
-        left: detail.x,
-        top: detail.y,
-        value: this.props.fields[type].map(f => f.property.id),
-        onInsert: (property) => {
-          this.props.onMetadataAdd({ id, property })
-        }
-      }
-    })
-  }
-
-  hideFieldPopup = () => {
-    this.setState({ fieldPopup: null })
+  handleFieldCreate = (property) => {
+    let id = this.props.edit.id[0]
+    this.props.onMetadataAdd({ id, property })
```

Comment:
We want to pass all the ids not just the first (if you're adding a field in the bulk editor you're adding it to all the items).

Reply:
We also don't want to pull the id out of `edit` here, that seems a bit dangerous. So we actually won't need `handleFieldCreate` here, just pass down `onMetadataAdd` to the lists' `onCreate` props directly.

We can handle create in `MetadataList` (adding `fields.id` as `id`).